core/src/test/scala/org/apache/spark/deploy/history/HistoryServerSuite.scala

```diff
@@ -212,6 +263,25 @@ class HistoryServerSuite extends FunSuite with BeforeAndAfter with Matchers with
     out.write(json)
     out.close()
   }
+
+  def unzipToDir(inputStream: InputStream, dir: File): Unit = {
+    val unzipStream = new ZipInputStream(inputStream)
+    val buffer = new Array[Byte](64 * 1024 * 1024)
+    var nextEntry = unzipStream.getNextEntry
+    while(nextEntry != null) {
+      val file = new File(dir, nextEntry.getName)
+      val outputStream = new BufferedOutputStream(new FileOutputStream(file))
+      var dataRemains = true
+      while (dataRemains) {
```

Comment:
Guava has `ByteStreams.copy()` which can be useful here (and probably in other places in this change?).